### ISSUE TITLE
Multiple DetailView: Alert messages displays always above first DetailView

### ISSUE THREAD

**Issue by vadim-bulochnik** [kartik-v/yii2-detail-view#95]
Hello,

When using several DetailView on one page - it's not possible to control place where alert messages are displays. For example, I have 3 DetailView on one page. If I updated last DetailView - I would like to see alert message above this view and not above first view on a page.

One of possible solutions: add possibility to set in alertMessageSettings (or in any other property) keys of flash messages that should be displayed for this DetailView (e.g. display only flash messages with keys 'kv-detail-myview-success' and 'kv-detail-myview-error').

Thank you,
Vadim

**Comment by kartik-v** [comment]
Alert messages are always displayed within each detail view. You need to control the navigation to a particular detail view by setting your detail view submit form action to point to the right bookmark on the page.

**Comment by vadim-bulochnik** [comment]
I've checked this problem again - and I can say that alert messages are always displayed only above first detail view.
The reason of this is quite clear: all detail views on a page uses the SAME keys for the flash messages. So, it's impossible to show alert message above particular detail view. Messages are always displayed above first detail view on a page.

Best regards,
Vadim

**Comment by kartik-v** [comment]
You may want to check [the documentation](http://demos.krajee.com/detail-view#settings) for understanding various settings for the widget.

For your use case, you can override the [alertMessageSettings property](http://demos.krajee.com/detail-view#option-alertMessageSettings) and set different session keys for different detail views you have on your page.

**Comment by kartik-v** [comment]
You may want to check [the documentation](http://demos.krajee.com/detail-view#settings) for understanding various settings for the widget.

For your use case, you can override the [alertMessageSettings property](http://demos.krajee.com/detail-view#option-alertMessageSettings) and set different session keys for different detail view widgets you have on your page.

For example if you have 3 detail views... you can set `kv-detail-error-1`, `kv-detail-error-2`, `kv-detail-error-3` as the session keys for the 3 detail views all pointing to say same error class `alert alert-error`.

**Comment by vadim-bulochnik** [comment]
Unfortunately, alertMessageSettings property not works correctly. 
For example, I have 2 detail views. I set following in first detail view:
`'alertMessageSettings' => [
            'kv-detail-success-1' => 'alert alert-success',
        ],`
And following in second detail view:
`'alertMessageSettings' => [
            'kv-detail-success-2' => 'alert alert-success',
        ],`

Then in controller I set flash message for the second detail view:
`Yii::$app->session->setFlash('kv-detail-success-2', Yii::t('app', 'Applicant information has been updated successfully.'));`

And result became worse: I still see message above first detail view, but with broken css. Message block (div) has class="alert alert-kv-detail-success-2 fade in"

Am I doing something wrong?
